packages/flutter_tools/lib/src/commands/build_ios.dart

```diff
@@ -117,14 +135,15 @@ class BuildIOSArchiveCommand extends _BuildIOSSubCommand {
             '"$exportOptions" is not a file. See "xcodebuild -h" for available keys.');
       }
     }
+    return super.validateCommand();
+  }
+
+  @override
+  Future<FlutterCommandResult> runCommand() async {
     final FlutterCommandResult xcarchiveResult = await super.runCommand();
     final BuildInfo buildInfo = await getBuildInfo();
     displayNullSafetyMode(buildInfo);
 
-    if (exportOptions == null) {
-      return xcarchiveResult;
-    }
-
     // xcarchive failed or not at expected location.
     if (xcarchiveResult.exitStatus != ExitStatus.success) {
       globals.printStatus('Skipping IPA');
```

Comment:
i assume the xcarchive failure details is already printed out?

Reply:
Yes, that's already done in `super.runCommand()`.